src/classes/OPP_CampaignMember_TDTM.cls

```diff
@@ -58,7 +58,7 @@ public class OPP_CampaignMember_TDTM extends TDTM_Runnable {
         
         dmlWrapper dmlWrapper = new DmlWrapper();
 
-        if (hasRun) {
+        if (recursionFlag > 0) {
```

Comment:
seems inconsistent to not create a separate class for maintaining the flag/counter, like you did for all the other components.